### ISSUE TITLE
Refactor CallImplicitGemmWrwDynamic()

### ISSUE THREAD

**Issue by atamazov** [ROCmSoftwarePlatform/MIOpen#855]
Depends on #691

> And `CallImplicitGemmWrwDynamic` will be refactored by a separate PR.

_Originally posted by @shaojiewang in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/691#issuecomment-816368176_

All the parameters that depend only on problem config should be computed in the Factory and passed to the Invoker (well, captured by it) by value.

If only one kernel us used, no need to use a vector of kernel objects. Manual adjustments of kernel time are not necessary in this case as well.

Example or refactoring: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/680/commits/6d9cf43ce3e7ee24fb6c63fe8c97ecd6fbd586d4

**Comment by shaojiewang** [comment]
It might be the same thing as #756 .
